### ISSUE TITLE
Tensor name "bn4b17_branch2c/moving_mean" not found in checkpoint files

### ISSUE THREAD

**Issue by StevenVerne** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#19]
I used the pretrained Resnet-101 provided from the tensorflow as you told in 11/09/2017 update
(http://download.tensorflow.org/models/resnet_v1_101_2016_08_28.tar.gz), but an NotFoundError is raised when load the model, as showed below:

NotFoundError (see above for traceback): Tensor name "bn4b17_branch2c/moving_mean" not found in checkpoint files G:/DeepLab/reference_model/tensorflow_official/resnet_v1_101.ckpt
	 [[Node: save_1/RestoreV2_202 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save_1/Const_0_0, save_1/RestoreV2_202/tensor_names, save_1/RestoreV2_202/shape_and_slices)]]
	 [[Node: save_1/RestoreV2_242/_183 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_1226_save_1/RestoreV2_242", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

I checked the downloaded "resnet_v1_101.ckpt" files, and found the tensor name in the ckpt is like 

tensor_name:  resnet_v1_101/block3/unit_5/bottleneck_v1/conv2/BatchNorm/moving_variance
tensor_name:  resnet_v1_101/block3/unit_14/bottleneck_v1/conv3/BatchNorm/gamma
tensor_name:  resnet_v1_101/block1/unit_3/bottleneck_v1/conv1/weights
tensor_name:  resnet_v1_101/block1/unit_3/bottleneck_v1/conv3/weights
tensor_name:  resnet_v1_101/block3/unit_16/bottleneck_v1/conv1/BatchNorm/gamma
tensor_name:  resnet_v1_101/block3/unit_7/bottleneck_v1/conv3/BatchNorm/gamma
tensor_name:  resnet_v1_101/block3/unit_18/bottleneck_v1/conv2/BatchNorm/gamma
tensor_name:  resnet_v1_101/block3/unit_7/bottleneck_v1/conv1/BatchNorm/beta
tensor_name:  resnet_v1_101/block3/unit_18/bottleneck_v1/conv2/weights

which is exactly different from what is needed.
Is there any process needed to be done before restoring the mode? or have I used a wrong model? Help

**Comment by John1231983** [comment]
Hi. Did you change the encoder name to res101? I guess you are using deeplab encoder

**Comment by StevenVerne** [comment]
I want to use the deeplab, should I run the res101 first?

P.S. I download the ckpt provided by DrSleep, the code works

**Comment by John1231983** [comment]
No. Just use encoder is deeplab and pretrain is deeplab_int that pretrained in MSCOCO. That's all. If you want to run encoder is resnet, the pretrain must be resnet_v1_101. The reason is that they must be same name to use pretrained weight. One more thing, the performance using resnet pretrained 70.5% is much lower than deeplab pretrained (by author), 75.5%. I did not find the problem yet. If you find it, lets me know

**Comment by StevenVerne** [comment]
Thank you for your advise. I am just starting to use the deeplab. Happy to share, if anything interesting found

**Comment by John1231983** [comment]
@zhengyang-wang: I think the convolution in the fc layer must be initial weight (such as Gaussian), otherwise the training from resnet pretrain will be very low.Do you think so?

**Comment by zhengyang-wang** [comment]
The architecture for the decoder part (named 'fc' in deeplab and 'decoder' in res101/50) is different from the original network (e.g. ASPP instead of fully connected layers). The number of parameters is different. We cannot transfer weights in this case. That's why we randomly initialized them.

**Comment by John1231983** [comment]
@zhengyang-wang: Hi. In your convolution weight, it is

```w = tf.get_variable('weights', shape=[kernel_size, kernel_size, num_x, num_o])```
I did not find the `initializer` flag is used. Does its default as random initialization? In my opition, I think it should be 

```w = tf.get_variable('weights', shape=[kernel_size, kernel_size, num_x, num_o],initializer= tf.contrib.layers.xavier_initializer())```

**Comment by zhengyang-wang** [comment]
For convolutional layers, we used pre-trained weights as their initialization. It does not matter what initializers you are using.

**Comment by zhengyang-wang** [comment]
And, the default initializer is https://www.tensorflow.org/api_docs/python/tf/glorot_uniform_initializer

**Comment by John1231983** [comment]
@zhengyang-wang : Thanks for your information. Actually, my question about convolutions is in the fc layer (decoder layers-ASPP). We have to initialize weight for these convolutions because we cannot borrow the weight from ResNet pre-train.
